tests/test_request_state.py

```diff
@@ -0,0 +1,30 @@
+from starlette.middleware.base import BaseHTTPMiddleware
+
+
+class StateMiddleware(BaseHTTPMiddleware):
+
+    async def dispatch(self, request, call_next):
+        request.state.test1 = int(12)
+        request.state.test2 = "Foo"
+
+        response = await call_next(request)
+        return response
+
+
+def test_api_request_state(api, url):
+    
+    api.add_middleware(StateMiddleware)
+
+    @api.route("/")
+    def home(req, resp):
+        assert hasattr(req, "state")
+        assert hasattr(req.state, "test1")
+        assert hasattr(req.state, "test2")
+        assert isinstance(req.state.test1, int)
+        assert isinstance(req.state.test2, str)
```

Comment:
`assert request.state.test1 == 12` and `assert request.state.test2 == "foo"` are enough no need for the `hasattr` or `isinstance`

Reply:
I don't think having the tests inside the home view is a good idea, you can simply return the state values in `resp.text` and do the test